### PR TITLE
TapArea: Add default cursor to TapArea

### PR DESCRIPTION
**Pull request by rlingineni**
### Summary
Adding default cursor support to TapArea

#### What changed?
Adding a default as a cursor option for hover only TapArea events

#### Why?
See context here in this [Slack](https://pinterest.slack.com/archives/C0HUV5J93/p1701286417650109)

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-XXXX)
- [TDD](link to Paper doc)
- [Figma](link to Figma file)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

**Comment by AlbertCarreras** [comment]
Did we come to a conclusion with this beyond needing some research?

**Comment by rlingineni** [comment]
@AlbertCarreras - yes, take a look at the the thread, and Cintia's research in the #Gestalt-Team channel

**Comment by diyorbek** [comment]
Looked through the thread. Makes sense to me!

**Review by AlbertCarreras — CHANGES_REQUESTED** [review]
I'm ok with merging after we consider this option: 


I think the case might make sense, and be valid, but by supporting this we open the box to a new world of behavior inconsistencies.
I've noticed that disabled IconButton + Tooltip still shows the Tooltip while showing arrow pointer.
TapArea behaves the same. I wonder if a workaround to achieve this is to use the disable state rather than a new pointer in Taparea as a way to ensure misusages of arrow pointer in actionable TapAreas Like in this [codesandbox example](https://codesandbox.io/p/sandbox/confident-cherry-p2fghk?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clpr2cao300063b6dk1k0qk9n%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clpr2cao300023b6d7ub97ekg%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clpr2cao300033b6d00xnwtyg%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clpr2cap8000i3b6do6s5t1ig%2522%257D%252C%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clpr2cao300053b6d5mqhzjp8%2522%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clpr2cap8000i3b6do6s5t1ig%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522SANDBOX_ONBOARDING%2522%252C%2522id%2522%253A%2522sandbox-onboarding%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%252C%2522id%2522%253A%2522clpr2cap8000i3b6do6s5t1ig%2522%252C%2522activeTabId%2522%253A%2522sandbox-onboarding%2522%257D%252C%2522clpr2cao300023b6d7ub97ekg%2522%253A%257B%2522id%2522%253A%2522clpr2cao300023b6d7ub97ekg%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clpr2cao300053b6d5mqhzjp8%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clpr2cao300043b6dbhc8jceh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%2522%257D%255D%252C%2522id%2522%253A%2522clpr2cao300053b6d5mqhzjp8%2522%252C%2522activeTabId%2522%253A%2522clpr2cao300043b6dbhc8jceh%2522%257D%252C%2522clpr2cao300033b6d00xnwtyg%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clpr2cao300033b6d00xnwtyg%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D). 

 Thoughts on this last option? (edited)


https://pinterest.slack.com/archives/C014X9LTRCN/p1701703499059089?thread_ts=1701371017.254999&cid=C014X9LTRCN

File: packages/gestalt/src/TapArea.js [hunk 1]
```diff
@@ -218,6 +227,7 @@ const TapAreaWithForwardRef: AbstractComponent<Props, HTMLDivElement> = forwardR
       [styles.fullHeight]: fullHeight,
       [styles.fullWidth]: fullWidth,
       [styles.copy]: mouseCursor === 'copy' && !disabled,
+      [styles.default]: mouseCursor === 'default' && !disabled,
```
Comment (AlbertCarreras) [inline packages/gestalt/src/TapArea.js:230]: && !disabled, is kind of redundant no? 

`      [styles.default]: mouseCursor === 'default' || disabled,
`
no?

File: packages/gestalt/src/TapArea.css [hunk 2]
```diff
@@ -6,6 +6,10 @@
   composes: xsCol12 from "./Column.css";
 }
 
+.default {
```
Comment (AlbertCarreras) [inline packages/gestalt/src/TapArea.css:9]: do we need this? Aint the current disabled state default to this already?

**Review by AlbertCarreras — APPROVED** [review]
with comments. added a few comments cos I'm unsure we need some of the extra logic but I might be wrong.

**Pull request merged**